harness/determined/common/storage/base.py

```diff
@@ -159,3 +162,40 @@ def _list_directory(root: Union[str, os.PathLike]) -> Dict[str, int]:
                 result[rel_path] = os.path.getsize(abs_path)
 
         return result
+
+
+def from_string(shortcut: str) -> StorageManager:
+    def process_query(q: Dict[str, List[str]]) -> Dict[str, str]:
+        return {k: v[0] for k, v in q.items()}
```

Comment:
If somebody gives a query string with duplicate parameter keys, we should throw an error rather than silently ignore one.